gui/src/ts/components/benchmark/BenchmarkAnswerPage.tsx

```diff
@@ -38,6 +56,251 @@ const QuestionAnswerChoiceCard: React.FunctionComponent<{
   </Card>
 );
 
+interface AnswerExplanationGraphNodeDatum extends ForceGraphNodeDatum {
+  paths: {
+    choiceAnalysisId: string;
+    questionAnswerPathId: string;
+    id: string;
+    score: number;
+  }[];
+  incomingEdges: number;
+  outgoingEdges: number;
+}
+
+interface AnswerExplanationGraphLinkDatum
+  extends ForceGraphLinkDatum<AnswerExplanationGraphNodeDatum> {
+  pathId: string;
+  questionAnswerPathId: string;
+  choiceAnalysisId: string;
+  score: number;
+}
+
+const answerExplanationGraphSimulation = d3
+  .forceSimulation<
+    AnswerExplanationGraphNodeDatum,
+    AnswerExplanationGraphLinkDatum
+  >()
+  .force(
+    "link",
+    d3
+      .forceLink<
+        AnswerExplanationGraphNodeDatum,
+        AnswerExplanationGraphLinkDatum
+      >()
+      .id((node) => node.id)
+    // .distance(100)
+    // .strength(1)
+  )
+  // .force("center", d3.forceCenter())
+  // .force("charge", d3.forceManyBody().strength(-300))
+  // .force("x", d3.forceX())
+  // .force("y", d3.forceY())
+  .force("collide", d3.forceCollide(50));
+
+const extractNodeAndLinks = (explanation: AnswerExplanation) => {
+  const choiceAnalyses = explanation.choiceAnalyses;
+
+  const nodes: {[nodeId: string]: AnswerExplanationGraphNodeDatum} = {};
+  const links: {[linkId: string]: AnswerExplanationGraphLinkDatum} = {};
+
+  choiceAnalyses
+    // ?.slice(0, 1)
+    ?.forEach(({questionAnswerPaths, choiceLabel}) => {
+      const choiceAnalysisId = choiceLabel;
+
+      questionAnswerPaths.forEach(({paths, endNodeId, startNodeId}) => {
+        const questionAnswerPathId = `${choiceAnalysisId}-${startNodeId}-${endNodeId}`;
+
+        paths.forEach(({path, score}, index) => {
+          const pathId = `${questionAnswerPathId}-${index}`;
+
+          // Extract nodes from path [node link node link node]
+          for (let i = 0; i < path.length; i += 2) {
+            const nodeId = path[i];
+            if (!nodes[nodeId]) {
```

Comment:
Loops shouldn't happen. You're ignoring them if they do. Throw an exception instead? Or handle them?

Reply:
The `nodes` contains the nodes for all the paths, so its not just checking if another path already added the node